### PR TITLE
xfail/skip some flaky distributed tests

### PR DESCRIPTION
**Pull request by jcrist**
- xfail `test_blockwise_dataframe_io` on osx for hdf5
- skip all distributed tests on windows (the failure is due to a fixture
cleanup error that lives in `distributed.utils_test`. This isn't easy to
patch around, and `xfail` won't cut it).

Both of these should *not* be long term solutions, but will (hopefully)
get our CI green again.

### CODE REVIEW

File: dask/tests/test_distributed.py [hunk 1]
```diff
@@ -25,6 +26,21 @@
...
 
 
+# TODO: the fixture teardown for `cluster_fixture` is failing periodically with
+# a PermissionError on windows only (in CI). Since this fixture lives in the
+# distributed codebase and is nested within other fixtures we use, it's hard to
+# patch it from the dask codebase. And since the error is during fixture
+# teardown, an xfail won't cut it. As a hack, for now we skip all these tests
+# on windows.
+pytestmark = pytest.mark.skipif(
+    sys.platform == "win32",
```
Comment (jrbourbeau) [inline dask/tests/test_distributed.py:36]: Optional style nit: I wouldn't mind storing what platform is used in variables in `dask/compatibility.py` similar to what `distributed` does [here](https://github.com/dask/distributed/blob/cd2ce86979de76c77ae361e1e03d1f74d3efbf53/distributed/compatibility.py#L10-L12). I often find myself having to look up how to specify "what OS am I on?"

File: dask/tests/test_distributed.py [hunk 2]
```diff
@@ -343,6 +359,10 @@ def test_blockwise_dataframe_io(c, tmpdir, io, fuse, from_futures):
     pd = pytest.importorskip("pandas")
     dd = pytest.importorskip("dask.dataframe")
 
+    # TODO: this configuration is flaky on osx in CI
+    if from_futures is True and fuse is False and sys.platform == "darwin":
```
Comment (jrbourbeau) [inline dask/tests/test_distributed.py:363]: We've also seen this test fail when `fuse=None` (xref https://github.com/dask/dask/issues/8816#issuecomment-1083544957). Maybe let's just be conservative and always `xfail`, regardless of what `fuse` is

File: dask/tests/test_distributed.py [hunk 3]
```diff
@@ -343,6 +359,10 @@ def test_blockwise_dataframe_io(c, tmpdir, io, fuse, from_futures):
     pd = pytest.importorskip("pandas")
     dd = pytest.importorskip("dask.dataframe")
 
+    # TODO: this configuration is flaky on osx in CI
```
Comment (jrbourbeau) [inline dask/tests/test_distributed.py:362]: Optional, but linking back to the corresponding issue is handy 

```suggestion
    # TODO: this configuration is flaky on osx in CI
    # See https://github.com/dask/dask/issues/8816
```

File: dask/tests/test_distributed.py [hunk 4]
```diff
@@ -25,6 +26,21 @@
     cluster = partial(cluster, should_check_state=False)
 
 
+# TODO: the fixture teardown for `cluster_fixture` is failing periodically with
+# a PermissionError on windows only (in CI). Since this fixture lives in the
+# distributed codebase and is nested within other fixtures we use, it's hard to
+# patch it from the dask codebase. And since the error is during fixture
+# teardown, an xfail won't cut it. As a hack, for now we skip all these tests
+# on windows.
```
Comment (jrbourbeau) [inline dask/tests/test_distributed.py:34]: Similar comment here

```suggestion
# on windows.
# See https://github.com/dask/dask/issues/8877
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @jcrist. Only one request (the comment about `fuse`), the rest of my comments are just style nits -- feel free to accept / ignore as you see fit
